skimage/future/detect/setup.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python
+
+from skimage._build import cython
+
+import os
+import tempfile
+import shutil
+#from numpy.distutils.command.build_ext import build_ext
+from distutils.errors import CompileError, LinkError
+from Cython.Distutils import build_ext
+
+base_path = os.path.abspath(os.path.dirname(__file__))
+
+compile_flags = ['-fopenmp']
+link_flags = ['-fopenmp']
+
+code = """#include <omp.h>
+int main(int argc, char** argv) { return(0); }"""
+
+class Checker(build_ext):
+
+    def can_compile_link(self):
+
+        cc = self.compiler
+        fname = 'test.c'
+        cwd = os.getcwd()
+        tmpdir = tempfile.mkdtemp()
+
+        try:
+            os.chdir(tmpdir)
+            with open(fname, 'wt') as fobj:
+                fobj.write(code)
+            try:
+                objects = cc.compile([fname],
+                                     extra_postargs=compile_flags)
+            except CompileError:
+                return False
+            try:
+                # Link shared lib rather then executable to avoid
+                # http://bugs.python.org/issue4431 with MSVC 10+
+                cc.link_shared_lib(objects, "testlib",
+                                   extra_postargs=link_flags)
+            except (LinkError, TypeError):
+                return False
+        finally:
+            os.chdir(cwd)
+            shutil.rmtree(tmpdir)
+        return True
+
+    def build_extensions(self):
+        """ Hook into extension building to check compiler flags """
+
+        print '-----------------------'
```

Comment:
Here's the errant `print` call Python 3.x doesn't like - change to `print('-----------------------')` after adding the above import and this should work (or... we'll graduate to a new, more exciting error ;)